### PR TITLE
Factory pattern for request and database classes

### PR DESCRIPTION
**Pull request by thijsdezoete**
Hello,

First of all, thanks for your awesome toolkit. After using it for a while I got tired of importing a lot of request objects. I created a factory for request objects and databases. If there's anything you need from me to get this merged, please let me know. :)

Example bot in readme could be rewritten like this:

```python
from factory import KolFactory as F
from kol.Session import Session

# Login to the KoL servers.
s = Session()
s.login("myUserName", "myPassword")

# Get a list of your kmails and print them out.
r = F('GetMessagesRequest', s)
responseData = r.doRequest()
kmails = responseData["kmails"]
for kmail in kmails:
    print "Received kmail from %s (#%s)" % (kmail["userName"], kmail["userId"])
    print "Text: %s" % kmail["text"]
    print "Meat: %s" % kmail["meat"]
    for item in kmail["items"]:
        print "Item: %s (%s)" % (item["name"], item["quantity"])

# Use an old coin purse.
item = F('ItemDatabase').getItemFromName("old coin purse")
r = F('UseItemRequest', s, item["id"])
r.doRequest()

# Smash a titanium assault umbrella and print out the results.
item = F('ItemDatabase').getItemFromName("titanium assault umbrella")
r = F('PulverizeRequest', s, item["id"])
responseData = r.doRequest()
smashResults = responseData["results"]
print "After smashing the item you have received the following:"
for result in smashResults:
    print "%s (%s)" % (result["name"], result["quantity"])

# Now we logout.
s.logout()
```

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
@thijsdezoete - This pull request, and its factory pattern, goes against general Python conventions.
Python code values explicit imports over implicit, whenever possible. (In fact, that's the second line in [The Zen of Python](http://www.python.org/dev/peps/pep-0020/), ```Explicit is better than implicit.```)
You can't be lazy with your imports just because you don't want to type them out. (Although, you can also be lazy by using ```import *```, but again, that's frowned upon. Though that's still better than importing classes using their string names). 

Using explicit imports at the top (and using actual classes rather than their string names) has multiple benefits. It improves readability, but more importantly, it allows the use of various support tools, such as pylint, static error checking, IDE auto-completes, jumping to a declaration of a class just by clicking on it, etc.

In short, this is the wrong approach. I recommend for this PR to be closed.

**Comment by scelis** [comment]
I mostly agree and it is the reason I have not done anything with this pull request. I prefer explicit code over "stringly typed" code.

**Comment by thijsdezoete** [comment]
I agree with the rationale. Though, I had friends playing and they weren't very skilled with the whole programming part, so I had this factory built in to make it easier for them to make everything 'just work'. I should've closed this a long time ago already. Sorry for not closing this earlier!
:+1:

**Comment by scelis** [comment]
I appreciate you sharing the code and opening the pull request, even if I don't agree with the direction of the code itself. :)
